libutils/hash_map.c

```diff
@@ -39,8 +44,8 @@ HashMap *HashMapNew(MapHashFn hash_fn, MapKeyEqualFn equal_fn,
     HashMap *map = xcalloc(1, sizeof(HashMap));
     map->hash_fn = hash_fn;
     map->equal_fn = equal_fn;
-    map->destroy_key_fn = destroy_key_fn;
-    map->destroy_value_fn = destroy_value_fn;
+    map->destroy_key_fn = (destroy_key_fn != NULL) ? destroy_key_fn : DestroyNothing;
```

Comment:
Would it not be better - where ever the `destroy_key_fn` and `destroy_value_fn` callbacks are called - to check them for `NULL`? E.g.
```
if (map->destroy_key_fn != NULL)
{
    map->destroy_key_fn(args...)
}
```
Calling an empty function is a lot more expensive then adding some `if`s here and there.